pinot-common/src/main/proto/request.proto

```diff
@@ -0,0 +1,128 @@
+/**
```

Comment:
we had a similar structure with thrift and found that it's limiting and makes operations/upgrades harder. 

let's go with something very simple and generic
```
message request {
  map<string,string> request_header;
  string sql; //optional
  repeated string segments;
  repeated byte payload; 
}
```